articles/stream-analytics/stream-analytics-parsing-json.md

```diff
@@ -110,42 +132,158 @@ FROM input as event
+Here is example of a single event. Both `CustomSensor03` and `SensorMetadata` are of type **array**:
+
+```json
+{
+    "DeviceId" : "12345",
+    "SensorReadings" :
+    {
+        "Temperature" : 80,
+        "Humidity" : 70,
+        "CustomSensor01" : 5,
+        "CustomSensor02" : 99,
+        "CustomSensor03": [12,-5,0]
+     },
+    "SensorMetadata":[
+        {          
+            "smKey":"Manufacturer",
+            "smValue":"ABC"                
+        },
+        {
+            "smKey":"Version",
+            "smValue":"1.2.45"
+        }
+    ]
+}
+```
 
 ### Working with a specific array element
+
 Select array element at a specified index (selecting the first array element):
 
 ```SQL
 SELECT
-    GetArrayElement(arrayField, 0) AS firstElement
+    GetArrayElement(SensorReadings.CustomSensor03, 0) AS firstElement
 FROM input
 ```
 
+The result is:
+
+|firstElement|
+|-|
+|12|
+
 ### Select array length
 
 ```SQL
 SELECT
-    GetArrayLength(arrayField) AS arrayLength
+    GetArrayLength(SensorReadings.CustomSensor03) AS arrayLength
 FROM input
 ```
 
+The result is:
+
+|arrayLength|
+|-|
+|3|
+
 ### Convert array elements into separate events
+
 Select all array element as individual events. The [APPLY](https://docs.microsoft.com/stream-analytics-query/apply-azure-stream-analytics) operator together with the [GetArrayElements](https://docs.microsoft.com/stream-analytics-query/getarrayelements-azure-stream-analytics) built-in function extracts all array elements as individual events:
 
 ```SQL
 SELECT
-    arrayElement.ArrayIndex,
-    arrayElement.ArrayValue
-FROM input as event
-CROSS APPLY GetArrayElements(event.arrayField) AS arrayElement
+    DeviceId,
+	CustomSensor03Record.ArrayIndex,
+	CustomSensor03Record.ArrayValue
+FROM input
+CROSS APPLY GetArrayElements(SensorReadings.CustomSensor03) AS CustomSensor03Record
+
 ```
 
+The result is:
+
+|DeviceId|ArrayIndex|ArrayValue|
+|-|-|-|
+|12345|0|12|
+|12345|1|-5|
+|12345|2|0|
+
+```SQL
+SELECT   
+    i.DeviceId,	
+    SensorMetadataRecords.ArrayValue.smKey as smKey,
+    SensorMetadataRecords.ArrayValue.smValue as smValue
+FROM input i
+CROSS APPLY GetArrayElements(SensorMetadata) AS SensorMetadataRecords
+ ```
+ 
+The result is:
+
+|DeviceId|smKey|smValue|
+|-|-|-|
+|12345|Manufacturer|ABC|
+|12345|Version|1.2.45|
+
+Using [WITH](https://docs.microsoft.com/en-us/stream-analytics-query/with-azure-stream-analytics), if necessary it's then possible to [join](https://docs.microsoft.com/en-us/stream-analytics-query/join-azure-stream-analytics) those fields in the main query (since this is not a join on reference data, it needs a [time boundary](https://docs.microsoft.com/en-us/stream-analytics-query/join-azure-stream-analytics#BKMK_DateDiff) that prevents duplication).
```

Comment:
This line is a little confusing. Can you please clarify?

Reply:
It is confusing indeed :)

The use case here is wanting to have Manufacturer and Version in columns instead of row. 
You will need to pivot the dataset in order to achieve that. 

One approach to pivoting is to join the processed dataset (where you have extracted your values in rows) with the original one (with the right cardinality of 1 row per incoming event). 

To do that in ASA you'll need to use 2 syntax elements: 

- the WITH clause, because you need to join the processed dataset with the original one as many times as columns you want to extract
- the JOIN clause. For that JOIN, you will need to define a time boundary in the join conditions (`DATEDIFF(...) BETWEEN ...`) because it's joining 2 live datasets. I wanted to make a note of that because it's not the case in the example give a bit earlier that was joining a reference dataset

Reply:
@field Thanks for contributing to Docs! Please remove `en-us` from the 3 links. Then those will work in all locales. ~Jason